src/main/java/gregtech/api/unification/material/info/MaterialIconType.java

```diff
@@ -117,16 +115,12 @@ public ResourceLocation getBlockTexturePath(@Nonnull MaterialIconSet materialIco
         }
 
         MaterialIconSet iconSet = materialIconSet;
-        //noinspection ConstantConditions
-        if (!iconSet.isRootIconset && FMLCommonHandler.instance().getEffectiveSide().isClient() &&
-                Minecraft.getMinecraft() != null) { // check minecraft for null for CI environments
-            IResourceManager manager = Minecraft.getMinecraft().getResourceManager();
+        if (!iconSet.isRootIconset && FMLCommonHandler.instance().getEffectiveSide().isClient()) {
             while (!iconSet.isRootIconset) {
-                try {
-                    // check if the texture file exists
-                    manager.getResource(new ResourceLocation(GTValues.MODID, String.format("textures/blocks/material_sets/%s/%s.png", iconSet.name, this.name)));
+                ResourceLocation location = new ResourceLocation(GTValues.MODID, String.format("textures/blocks/material_sets/%s/%s.png", iconSet.name, this.name));
+                if (ResourceHelper.doResourcepacksHaveResource(location)) {
                     break;
-                } catch (IOException ignored) {
+                } else {
                     iconSet = iconSet.parentIconset;
                 }
             }
```

Comment:
This looks like nearly same as block bellow, except texture path. Does it make sense to extract it out?